### PR TITLE
Task 'Calculate' solved with std::map and function pointer

### PR DESCRIPTION
**Pull request by gitbart74859**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: calculate
🏅 Granted 6 XP for @gitbart74859

File: homework/calculate/calculate.hpp [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+#include <iostream>
 #include <string>
+#include <functional>
+#include <map>
+#include <string>
+
+
+// std::map accepts only types of parameters in template barckets,
+// therefore 'typedef' must be used to declere function pointer type
+typedef std::string (*functionType)(int first, int second);
```
Comment (nauka-programowania-MA) [inline homework/calculate/calculate.hpp:11]: maybe using? typedef is deprecated in most situations

File: homework/calculate/calculate.hpp [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+
+std::map<std::string, functionType> commands
+{
+    {"add", [](int first, int second){return std::to_string(first + second);}},
+    {"subtract", [](int first, int second){return std::to_string(first - second);}},
+    {"multiply", [](int first, int second){return std::to_string(first * second);}},
+    {"divide", [](int first, int second){
+        if(second == 0)
+        {
+             //Returning types should be consistent
```
Comment (nauka-programowania-MA) [inline homework/calculate/calculate.hpp:21]: redundant comment return string describe problem

**Review by nauka-programowania-MA — COMMENTED** [review]
Nice!

**Pull request closed without merging**